sound/soc/sof/ipc4-pcm.c

```diff
@@ -835,5 +835,6 @@ const struct sof_ipc_pcm_ops ipc4_pcm_ops = {
 	.dai_link_fixup = sof_ipc4_pcm_dai_link_fixup,
 	.pcm_setup = sof_ipc4_pcm_setup,
 	.pcm_free = sof_ipc4_pcm_free,
-	.delay = sof_ipc4_pcm_delay
+	.delay = sof_ipc4_pcm_delay,
+	.trigger_ipc_first_on_start = true
```

Comment:
something doesn't add up in the commit message

"But with IPC4, FW programs the
DMA when the SET_PIPELINE_STATE IPC is sent during sof_pcm_trigger() and
this IPC is sent after the host sets the DMA RUN bit in the same
function."

If the IPC and DMA RUN happen in the same sof_pcm_trigger() function, it's not clear what changing the order of snd_sof_pcm_platform_trigger() and pcm_ops->trigger() might do?

Reply:
@plbossart IPC is sent in pcm_ops->trigger().

Reply:
@ranj063 possibly but then your commit message is not quite right. The "same function" refers to sof_pcm_trigger() with conventional grammar.

Reply:
@plbossart I'm not sure I understand your concern. The grammar is correct and I do mean same function as in sof_pcm_trigger(). But within the same function, we do 2 things: send ipc and set DMA run but. And these are done in pcm_ops->trigger and platform_pcm_trigger(). And the ordering between the two is what needs changing for ipc4. Does that make sense?

Reply:
It's not clear what 'same' means when you are referring to three functions.

You can reword to remove the ambiguity without using 'same', e.g.

With IPC4, the sof_pcm_trigger() functions performs two operations in incorrect order.

It firsts request to send an IPC pcm_ops->trigger and later programs the DMA RUN with platform_pcm_trigger().

This patch reverts the order of the two operations for IPC4 only.